docs/architecture.md

```diff
@@ -82,17 +94,32 @@ def my_command(ctx, pulp_ctx):
     ctx.obj = MyEntityContext(pulp_ctx)
 ```
 
-The named tuple `PluginRequirement` is used to describe dependence on server components.
-It needs the `name` of the plugin and accepts optionally an inclusive `min` version,
-an exclusive `max` version, and an `inverted` flag for exclusion of a version range.
+To declare version restrictions on *options* (*body parameters* in the API), the [`preprocess_entity`][pulp_glue.common.context.PulpEntityContext.preprocess_entity]
+method can check if a given option is being used and conditionally apply the requirements to the context.
+
+Each *plugin* may already have `Contexts` binded to specific [resource_class](/using_the_cli/#general-command-syntax) entities,
+which can be found under `pulp_glue/{plugin-name}/context.py`. In the example, a guard is added to `my_option`, which belongs to the
+`Repository` *resource_class* of `"my_plugin"`.
```

Comment:
This is a bit unclear to me. Whenever you are talking to an entity, there should most definitely be a context for it.

Reply:
I see. The relation about entities and context was not very clear. How about (without the bold)
> Each *plugin*  **have** `Contexts` binded to specific [resource_class](/using_the_cli/#general-command-syntax) entities, which can be found under `pulp_glue/{plugin-name}/context.py`. (...)

Reply:
better for sure.